pkg/registry/apis/datasource/sub_query.go

```diff
@@ -77,21 +80,27 @@ func (r *subQueryREST) Connect(ctx context.Context, name string, opts runtime.Ob
 	}
 
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		queries, err := r.readQueries(req)
+		queries, dsRef, err := r.readQueries(req)
 		if err != nil {
 			responder.Error(err)
 			return
 		}
+		if dsRef != nil && dsRef.UID != name {
+			responder.Error(fmt.Errorf("expected the datasource in the request url and body to match"))
+			return
+		}
 
 		queryResponse, err := r.builder.client.QueryData(ctx, &backend.QueryDataRequest{
 			PluginContext: pluginCtx,
 			Queries:       queries,
 			//  Headers: // from context
 		})
 		if err != nil {
+			responder.Error(err)
 			return
 		}
 
+		// TODO: get format (maybe protobuf!!!)
```

Comment:
TBD -- for machine-machine communication, the protobuf format may be a better option than JSON.  

For the grpc based plugins, it will absolutely be better/more efficient if we can just pass through the raw protobuf :) rather than converting it from `protobuf > arrow > go structs > json`

Reply:
I added a much bigger comment to explain why I think it is important